pkg/apis/application/v1alpha1/types.go

```diff
@@ -97,6 +97,16 @@ func (e Env) Environ() []string {
 	return environ
 }
 
+// does an operation similar to `envstubst` tool
+// see https://linux.die.net/man/1/envsubst
+func (e Env) Envsubst(s string) string {
+	for _, e := range e {
+		s = strings.ReplaceAll(s, fmt.Sprintf("$%s", e.Name), e.Value)
+		s = strings.ReplaceAll(s, fmt.Sprintf("${%s}", e.Name), e.Value)
+	}
```

Comment:
Hm, what about references in `s` to variables that are not set? `envsubst` in Linux silently discards them (i.e. replace them by the empty string `""` , do we want to mimic that behaviour?

Reply:
interesting question, I think this would always be a problem, which you'd need to fix

Reply:
I've decided not to change this. Fix is complex.